### PR TITLE
User-class,-authentication-etc

### PR DESCRIPTION
**Pull request by wicksa-art**

### CODE REVIEW

File: src/main/java/lv/startup/BalticWaterTemp/core/entity/User.java [hunk 1]
```diff
@@ -1,26 +1,31 @@
...
 @AllArgsConstructor
-@Data
-public class User {
+@Entity
+@Table(name="users")
+public class User
+{
+
+    @Column(nullable=false)
+    private String name;
```
Comment (321Dimak) [inline src/main/java/lv/startup/BalticWaterTemp/core/entity/User.java:19]: Add column name to all fields, and you can remove nullable and unique it is set in database

File: src/main/java/lv/startup/BalticWaterTemp/core/security/dto/UserDto.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+public class UserDto
+{
+    @NotEmpty
```
Comment (321Dimak) [inline src/main/java/lv/startup/BalticWaterTemp/core/security/dto/UserDto.java]: Why here no message?

File: src/main/java/lv/startup/BalticWaterTemp/core/services/UserService.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import lv.startup.BalticWaterTemp.core.security.dto.UserDto;
+
+import java.util.List;
+
+public interface UserService {
+    void saveUser(UserDto userDto);
+
+    User findByEmail(String email);
+
+    List<UserDto> findAllUsers();
```
Comment (321Dimak) [inline src/main/java/lv/startup/BalticWaterTemp/core/services/UserService.java:13]: I think we don't need this method

File: src/main/java/lv/startup/BalticWaterTemp/core/services/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+    private JpaUserRepository JpaUserRepository;
+    private PasswordEncoder passwordEncoder;
+
+    public UserServiceImpl(JpaUserRepository userRepository,
```
Comment (321Dimak) [inline src/main/java/lv/startup/BalticWaterTemp/core/services/impl/UserServiceImpl.java]: You don't need constructor use @Autowired

File: src/main/java/lv/startup/BalticWaterTemp/core/services/impl/UserServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+    private JpaUserRepository JpaUserRepository;
```
Comment (321Dimak) [inline src/main/java/lv/startup/BalticWaterTemp/core/services/impl/UserServiceImpl.java]: Use @Autowired

File: src/main/java/lv/startup/BalticWaterTemp/core/services/impl/UserServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    @Override
+    public void saveUser(UserDto userDto) {
+        User user = new User();
+        user.setName(userDto.getName());
+        user.setEmail(userDto.getEmail());
+
+        //encrypt the password once we integrate spring security
+        //user.setPassword(userDto.getPassword());
+        user.setPassword(passwordEncoder.encode(userDto.getPassword()));
+        JpaUserRepository.save(user);
```
Comment (321Dimak) [inline src/main/java/lv/startup/BalticWaterTemp/core/services/impl/UserServiceImpl.java]: I think you can do all this method in one line

File: src/main/java/lv/startup/BalticWaterTemp/core/services/impl/UserServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    @Override
+    public User findByEmail(String email) {
+        return JpaUserRepository.findByEmail(email);
+    }
+
+    @Override
```
Comment (321Dimak) [inline src/main/java/lv/startup/BalticWaterTemp/core/services/impl/UserServiceImpl.java:30]: we dont need this

File: src/main/java/lv/startup/BalticWaterTemp/core/services/impl/UserServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    public List<UserDto> findAllUsers() {
+        List<User> users = JpaUserRepository.findAll();
+        return users.stream().map((user) -> convertEntityToDto(user))
+                .collect(Collectors.toList());
+    }
+
+    private UserDto convertEntityToDto(User user){
+        UserDto userDto = new UserDto();
+        String[] name = user.getName().split(" ");
```
Comment (321Dimak) [inline src/main/java/lv/startup/BalticWaterTemp/core/services/impl/UserServiceImpl.java]: why you split name?

File: src/main/java/lv/startup/BalticWaterTemp/core/services/impl/UserServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+        userDto.setName(name[0]);
+        userDto.setEmail(user.getEmail());
+        return userDto;
```
Comment (321Dimak) [inline src/main/java/lv/startup/BalticWaterTemp/core/services/impl/UserServiceImpl.java]: and if you don't need to split name, you can do it in one line

File: src/main/java/lv/startup/BalticWaterTemp/core/security/CustomUserDetailsService.java [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+import lv.startup.BalticWaterTemp.core.entity.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+
+    private JpaUserRepository JpaUserRepository;
```
Comment (321Dimak) [inline src/main/java/lv/startup/BalticWaterTemp/core/security/CustomUserDetailsService.java]: use @Autowired

File: src/main/java/lv/startup/BalticWaterTemp/core/security/CustomUserDetailsService.java [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+
+    public CustomUserDetailsService(JpaUserRepository userRepository) {
```
Comment (321Dimak) [inline src/main/java/lv/startup/BalticWaterTemp/core/security/CustomUserDetailsService.java]: you don't need constructor

File: src/main/java/lv/startup/BalticWaterTemp/web_ui/controllers/AuthController.java [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
+
+import java.util.List;
+
+@Controller
+public class AuthController {
+
+    private UserService userService;
```
Comment (321Dimak) [inline src/main/java/lv/startup/BalticWaterTemp/web_ui/controllers/AuthController.java]: @Autowired

File: src/main/java/lv/startup/BalticWaterTemp/web_ui/controllers/AuthController.java [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+
+    public AuthController(UserService userService) {
```
Comment (321Dimak) [inline src/main/java/lv/startup/BalticWaterTemp/web_ui/controllers/AuthController.java]: don't need it

**Review by 321Dimak — CHANGES_REQUESTED** [review]

File: src/test/java/lv/startup/BalticWaterTemp/BalticWaterTempApplicationTests.java [hunk 14]
```diff
@@ -7,8 +7,5 @@
 @SpringBootTest
 class BalticWaterTempApplicationTests {
 
-	@Test
-	void contextLoads() {
-	}
```
Comment (321Dimak) [inline src/test/java/lv/startup/BalticWaterTemp/BalticWaterTempApplicationTests.java:9]: Why you removed test

**Review by 321Dimak — CHANGES_REQUESTED** [review]

File: src/test/java/lv/startup/BalticWaterTemp/BalticWaterTempApplicationTests.java (hunk 14 above)
Comment (wicksa-art) [reply src/test/java/lv/startup/BalticWaterTemp/BalticWaterTempApplicationTests.java:9]: I don't see that I made this change in Github desktop

**Review by 321Dimak — APPROVED** [review]

**Pull request merged**